webhooks-extension/pkg/endpoints/sink.go

```diff
@@ -319,6 +428,52 @@ func definePipelineRun(pipelineRunName, namespace, saName, repoURL, branch strin
 	return pipelineRunPointer, nil
 }
 
+/* Create a new TaskRun - repoUrl, resourceBinding and params can be nill depending on the Pipeline
+each TaskRun has a 1 hour timeout: */
+func defineTaskRun(taskRunName, namespace, saName, repoURL string, branch string,
+	task v1alpha1.Task,
+	resourceBinding []v1alpha1.TaskResourceBinding,
+	params []v1alpha1.Param) (*v1alpha1.TaskRun, error) {
+
+	gitServer, gitOrg, gitRepo := "", "", ""
+	err := errors.New("")
+	if repoURL != "" {
+		gitServer, gitOrg, gitRepo, err = getGitValues(repoURL)
+		if err != nil {
+			logging.Log.Errorf("error getting the Git values: %s.", err)
+			return &v1alpha1.TaskRun{}, err
+		}
+	}
+
+	taskRunData := v1alpha1.TaskRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      taskRunName,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"app":          "tekton-webhook-handler",
+				gitServerLabel: gitServer,
+				gitOrgLabel:    gitOrg,
+				gitRepoLabel:   gitRepo,
+				gitBranchLabel: branch,
+			},
+		},
+
+		Spec: v1alpha1.TaskRunSpec{
+			TaskRef:        &v1alpha1.TaskRef{Name: task.Name},
+			Timeout:        &metav1.Duration{Duration: 1 * time.Hour},
```

Comment:
Is there a possibility we might want this to be configurable? Possibly a follow on issue?

Reply:
Yes, I will address this in a follow up issue.